image_generation/common/diffusers/src/scheduler_lcm.cpp

```diff
@@ -0,0 +1,272 @@
+                           num_train_timesteps_config(num_train_timesteps),
+                           original_inference_steps_config(original_inference_steps),
+                           timestep_scaling_config(timestep_scaling),
+                           thresholding(thresholding),
+                           clip_sample(clip_sample),
+                           clip_sample_range(clip_sample_range),
+                           dynamic_thresholding_ratio(dynamic_thresholding_ratio),
+                           sample_max_value(sample_max_value),
+                           read_torch_noise(read_torch_noise) {
+
+    sigma_data = 0.5f; // Default: 0.5
+
+    std::vector<float> alphas, betas;
+
+    if (!trained_betas.empty()) {
+        auto betas = trained_betas;
+    } else if (beta_schedule == BetaSchedule::LINEAR) {
+        for (int32_t i = 0; i < num_train_timesteps; i++) {
+            betas.push_back(beta_start + (beta_end - beta_start) * i / (num_train_timesteps - 1));
+        }
+    } else if (beta_schedule == BetaSchedule::SCALED_LINEAR) {
+        float start = sqrt(beta_start);
+        float end = sqrt(beta_end);
+        std::vector<float> temp = linspace(start, end, num_train_timesteps);
+        for (float b : temp) {
+            betas.push_back(b * b);
+        }
+    } else {
+        OPENVINO_THROW("'beta_schedule' must be one of 'EPSILON' or 'SCALED_LINEAR'");
+    }
+    for (float b : betas) {
+        alphas.push_back(1.0f - b);
+    }
+    for (size_t i = 1; i <= alphas.size(); i++) {
+        float alpha_cumprod =
+            std::accumulate(std::begin(alphas), std::begin(alphas) + i, 1.0, std::multiplies<float>{});
+        alphas_cumprod.push_back(alpha_cumprod);
+    }
+
+    final_alpha_cumprod = set_alpha_to_one ? 1 : alphas_cumprod[0];
+}
+
+void LCMScheduler::set_timesteps(size_t num_inference_steps) {
+    float strength = 1.0f;
+    // LCM Timesteps Setting
+    size_t k = num_train_timesteps_config / original_inference_steps_config;
+
+    size_t origin_timesteps_size = original_inference_steps_config * strength;
+    std::vector<size_t> lcm_origin_timesteps(origin_timesteps_size);
+    std::iota(lcm_origin_timesteps.begin(), lcm_origin_timesteps.end(), 1);
+    std::transform(lcm_origin_timesteps.begin(), lcm_origin_timesteps.end(), lcm_origin_timesteps.begin(), [&k](auto& x) {
+        return x * k - 1;
+    });
+
+    size_t skipping_step = origin_timesteps_size / num_inference_steps;
+    assert(skipping_step >= 1 && "The combination of `original_steps x strength` is smaller than `num_inference_steps`");
+
+    this->num_inference_steps = num_inference_steps;
+    // LCM Inference Steps Schedule
+    std::reverse(lcm_origin_timesteps.begin(),lcm_origin_timesteps.end());
+
+    // v1. based on master branch: https://github.com/huggingface/diffusers/blame/2a7f43a73bda387385a47a15d7b6fe9be9c65eb2/src/diffusers/schedulers/scheduling_lcm.py#L387 
```

Comment:
```suggestion
    // v1. based on https://github.com/huggingface/diffusers/blame/2a7f43a73bda387385a47a15d7b6fe9be9c65eb2/src/diffusers/schedulers/scheduling_lcm.py#L387 
```
You are already specific enough

Reply:
Fixed